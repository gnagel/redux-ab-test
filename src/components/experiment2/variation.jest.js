import React from 'react'; // eslint-disable-line no-unused-vars
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import reduxAbTest, { initialState } from '../../module';
import Variation, { mapStateToProps } from './variation';

describe('src/components/experiments2/variation.js', () => {
  let component;
  let tree;

  describe('plain text', () => {
    beforeEach(() => {
      component = renderer.create(
        <Provider store={createStore(reduxAbTest, { reduxAbTest: initialState })}>
          <Variation name='Test variation'>
            Testing some text
          </Variation>
        </Provider>
      );
      tree = component.toJSON();
    });

    it('exists', () => {
      expect(component).not.toBeUndefined;
    });

    it('has the correct text', () => {
      expect(tree).toMatchSnapshot();
      expect(tree.children).toEqual(['Testing some text']);
    });

    it('has the correct tagName', () => {
      expect(tree).toMatchSnapshot();
      expect(tree.type).toEqual('span');
    });
  });

  describe('component children', () => {
    beforeEach(() => {
      component = renderer.create(
        <Provider store={createStore(reduxAbTest, { reduxAbTest: initialState })}>
          <Variation name='Test variation'>
            <div>Testing some text</div>
          </Variation>
        </Provider>
      );
      tree = component.toJSON();
    });

    it('exists', () => {
      expect(component).not.toBeUndefined;
    });

    it('has the correct text', () => {
      expect(tree).toMatchSnapshot();
      expect(tree.children).toEqual(['Testing some text']);
    });

    it('has the correct tagName', () => {
      expect(tree).toMatchSnapshot();
      expect(tree.type).toEqual('div');
    });
  });
});
