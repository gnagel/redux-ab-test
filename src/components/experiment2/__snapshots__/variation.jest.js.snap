// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/experiments2/variation.js component children has the correct tagName 1`] = `
<div>
  Testing some text
</div>
`;

exports[`src/components/experiments2/variation.js component children has the correct text 1`] = `
<div>
  Testing some text
</div>
`;

exports[`src/components/experiments2/variation.js plain text has the correct tagName 1`] = `
<span>
  Testing some text
</span>
`;

exports[`src/components/experiments2/variation.js plain text has the correct text 1`] = `
<span>
  Testing some text
</span>
`;
